pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -748,9 +748,9 @@ func TestAppProbe(t *testing.T) {
 					HTTPGet: &apimirror.HTTPGetAction{
 						Port: intstr.IntOrString{IntVal: int32(appPort)},
 						Path: "/header",
-						Host: testHostValue,
 						HTTPHeaders: []apimirror.HTTPHeader{
 							{Name: testHeader, Value: testHeaderValue},
+							{Name: "Host", Value: testHostValue},
```

Comment:
Adding this `Host` header and the below logic does not cause the existing code to fail. Could we add a comment in the tests describing why the `req.Host` assignment is necessary and what functionality it replicates?

Reply:
This is the right way to set `Host`, previously HTTPGet.Host is wrong

Reply:
I agree that it is the correct way to set `Host`. My concern is that this test doesn't ensure `Host` is set correctly. These tests still pass on the previous changes and therefore wouldn't catch a future regression.

Reply:
I am not super sure what you mean by regression